using-mobile-extensions/mobile-core/identity/identity-api-reference.md

```diff
@@ -111,32 +111,58 @@ If the provided URL is nil or empty, it is returned as is. Otherwise, the follow
 **Objective-C**
 
 ```objectivec
-NSURL* url = [[NSURL alloc] initWithString:@"www.myUrl.com"];
+NSURL* url = [[NSURL alloc] initWithString:@"https://example.com"];
 [ACPIdentity appendToUrl:url withCallback:^(NSURL * _Nullable urlWithVisitorData) {    
-// handle the appended url here}
+	// handle the appended url here
+	
+	// APIs which update the UI must be from main thread
+	dispatch_async(dispatch_get_main_queue(), ^{
+		[[self webView] loadRequest:[NSURLRequest requestWithURL:urlWithVisitorData]];
+  }
 }];
 
 [ACPIdentity appendToUrl:url withCompletionHandler:^(NSURL * _Nullable urlWithVersionData, NSError * _Nullable error) {
    if (error) {
      // handle error here
```

Comment:
handle error here, check ACPError for possible error codes that can be handled - same below

Reply:
I feel this is a slippery slope where the API example get muddled with attempting to handle every possible error. If we add new error codes to ACPError then we need to update all these examples. Instead, I've added documentation for ACPError and AdobeError in the Core API reference page and then linked to them from the Identity API reference page when they are mentioned.